core/src/main/scala/filodb.core/query/QueryUtils.scala

```diff
@@ -1,16 +1,25 @@
 package filodb.core.query
 
+import com.github.benmanes.caffeine.cache.{Cache, Caffeine}
 import scala.collection.mutable
 import scala.collection.mutable.ArrayBuffer
 
 /**
  * Storage for utility functions.
  */
 object QueryUtils {
-  val REGEX_CHARS = Array('.', '?', '+', '*', '|', '{', '}', '[', ']', '(', ')', '"', '\\')
+  val REGEX_CHARS: Array[Char] = Array('.', '?', '+', '*', '|', '{', '}', '[', ']', '(', ')', '"', '\\')
+  private val COMBO_CACHE_SIZE = 2048
```

Comment:
Should this be configurable, too?
Good catch btw -- I wonder if this lit up in your flame plots, too 🤔

Reply:
> and  QueryUtils.makeAllKeyValueCombos is high
Missed this bit 👏

Reply:
It's an Object and a util and doesn't have a config yet, passing it a config might be too much. I'll let it go as a constant for now and then we can extract a config if needed? I am open to have a different value.